services/processor/tasks/miner/miner.go

```diff
@@ -199,190 +199,6 @@ func (mac *ProcessMinerTask) Task(job *work.Job) error {
-		} else if !found {
-			panic("Undefined behaviour when a partition is removed.")
-		}
-		partitionDiff, err := diffPartition(node.Store(), prevPart, curPart)
-		if err != nil {
-			return err
-		}
-		out[uint64(i)] = partitionDiff
-
-		return nil
-	}); err != nil {
-		return nil, err
-	}
-	return out, nil
-}
-
-func diffPartition(store adt.Store, prevPart, curPart miner.Partition) (*minermodel.PartitionStatus, error) {
-	// all the sectors that were in previous but not in current
-	allRemovedSectors, err := bitfield.SubtractBitField(prevPart.Sectors, curPart.Sectors)
-	if err != nil {
-		return nil, err
-	}
-
-	// list of sectors that were terminated before their expiration.
-	terminatedEarlyArr, err := adt.AsArray(store, curPart.EarlyTerminated)
-	if err != nil {
-		return nil, err
-	}
-
-	expired := bitfield.New()
-	var bf bitfield.BitField
-	if err := terminatedEarlyArr.ForEach(&bf, func(i int64) error {
-		// expired = all removals - termination
-		expirations, err := bitfield.SubtractBitField(allRemovedSectors, bf)
-		if err != nil {
-			return err
-		}
-		// merge with expired sectors from other epochs
-		expired, err = bitfield.MergeBitFields(expirations, expired)
-		if err != nil {
-			return nil
-		}
-		return nil
-	}); err != nil {
-		return nil, err
-	}
-
-	// terminated = all removals - expired
-	terminated, err := bitfield.SubtractBitField(allRemovedSectors, expired)
-	if err != nil {
-		return nil, err
-	}
-
-	// faults in current but not previous
-	faults, err := bitfield.SubtractBitField(curPart.Recoveries, prevPart.Recoveries)
-	if err != nil {
-		return nil, err
-	}
-
-	// recoveries in current but not previous
-	inRecovery, err := bitfield.SubtractBitField(curPart.Recoveries, prevPart.Recoveries)
-	if err != nil {
-		return nil, err
-	}
-
-	// all current good sectors
-	newActiveSectors, err := curPart.ActiveSectors()
-	if err != nil {
-		return nil, err
-	}
-
-	// sectors that were previously fault and are now currently active are considered recovered.
-	recovered, err := bitfield.IntersectBitField(prevPart.Faults, newActiveSectors)
-	if err != nil {
-		return nil, err
-	}
-
-	return &minermodel.PartitionStatus{
-		Terminated: terminated,
-		Expired:    expired,
-		Faulted:    faults,
-		InRecovery: inRecovery,
-		Recovered:  recovered,
-	}, nil
-}
-
-func minerStateAt(ctx context.Context, node lens.API, maddr address.Address, tskey types.TipSetKey) (miner.State, error) {
-	prevActor, err := node.StateGetActor(ctx, maddr, tskey)
-	if err != nil {
-		return miner.State{}, err
-	}
-	var out miner.State
-	// Get the miner state info
-	astb, err := node.ChainReadObj(ctx, prevActor.Head)
-	if err != nil {
-		return miner.State{}, err
-	}
-	if err := out.UnmarshalCBOR(bytes.NewReader(astb)); err != nil {
-		return miner.State{}, err
-	}
-	return out, nil
+func (p *ProcessMinerTask) minerPartitionsDiff(ctx context.Context, prevState, curState miner.State) (map[uint64]*minermodel.PartitionStatus, error) {
+	panic("NYI")
```

Comment:
The logic of this method [was wrong](https://github.com/filecoin-project/lotus/pull/3781/files#diff-90b0abb6d8074578d3411fbe7bfe7d65R696-R701) (linked to large PR, see ` cmd/lotus-chainwatch/processor/miner.go` for details on wrongness) and will need to be re-implemented. This will come in a follow on.